src/django_components/component_media.py

```diff
@@ -953,27 +954,47 @@ class MyComp(Component):
     asset_content = getattr(comp_media, inlined_attr, None)
     asset_file = getattr(comp_media, file_attr, None)
 
-    if asset_file is not None:
-        # Check if the file is in one of the components' directories
-        full_path = resolve_file(asset_file, comp_dirs)
-
-        if full_path is None:
-            # If not, check if it's in the static files
-            if type == "static":
-                full_path = finders.find(asset_file)
-            # Or in the templates
-            elif type == "template":
-                try:
-                    template: Template = get_template(asset_file)
-                    full_path = template.origin.name
-                except TemplateDoesNotExist:
-                    pass
-
-        if full_path is None:
-            # NOTE: The short name, e.g. `js` or `css` is used in the error message for convenience
-            raise ValueError(f"Could not find {inlined_attr} file {asset_file}")
-
-        # NOTE: Use explicit encoding for compat with Windows, see #1074
-        asset_content = Path(full_path).read_text(encoding="utf8")
+    # No inlined content, nor file name
+    if asset_content is None and asset_file is None:
+        return None
+
+    if asset_content is not None and asset_file is not None:
+        raise ValueError(
+            f"Received both '{inlined_attr}' and '{file_attr}' in Component {comp_cls.__qualname__}."
+            " Only one of the two must be set."
+        )
+
+    # If the content was inlined into the component (e.g. `Component.template = "..."`)
+    # then there's nothing to resolve. Return as is.
+    if asset_content is not None:
+        return asset_content
+
+    # The rest of the code assumes that we were given only a file name
+    asset_file = cast(str, asset_file)
+
+    if type == "template":
+        # NOTE: While we return on the "source" (plain string) of the template,
+        #       by calling `load_component_template()`, we also cache the Template instance.
+        #       So later in Component's `render_impl()`, we don't have to re-compile the Template.
+        template = load_component_template(comp_cls, asset_file)
```

Comment:
Previously, we did use Django's `get_template()` to load the template, but then we discarded the actual `Template` instance, and read the file contents as plain string.

Now, one difference is that when calling `load_component_template()` here, we save the loaded `Template` instance to `Component._template`.

So we use the Component classes to cache the Template instances.

And because we now use the `Template` instance returned from Django's `get_template()`, we now play nicely with template loaders (https://github.com/django-components/django-components/issues/901)